### PR TITLE
Fix PACKET_equal test with BUF_LEN+1 on -Wstringop-overread

### PR DESCRIPTION
**Pull request by rurban**
Detected with -Werror.
Now the build and all tests pass through with asan, ubsan, -O2 and -Werror on gcc-11 and gcc-12.

Then we could add -Werror on all known compilers.

##### Checklist
- [x] tests are added or updated

### CODE REVIEW

**Comment by t8m** [comment]
This break CI.

**Comment by hlandau** [comment]
I actually encountered this and fixed this in a PR I'm working on, but did so by changing smbuf to be one byte longer. I think that would be a better approach.

**Comment by rurban** [comment]
@hlandau: Agreed `static unsigned char smbuf[BUF_LEN + 1];` would be better.
And I had a wrong gcc version check. It's gcc 10, not 3.10.

Using now hlandau's idea

**Review by hlandau — APPROVED** [review]

**Review by t8m — APPROVED** [review]

**Comment by t8m** [comment]
OK with CLA trivial

**Comment by t8m** [comment]
@hlandau do you agree with CLA: trivial?

**Comment by t8m** [comment]
@hlandau ping, are you OK with CLA trivial?

File: test/packettest.c [hunk 1]
```diff
@@ -12,13 +12,13 @@
 
 #define BUF_LEN 255
 
-static unsigned char smbuf[BUF_LEN];
+static unsigned char smbuf[BUF_LEN + 1];
```
Comment (slontis) [inline test/packettest.c:15]: It might have been nice to say why this was done with a /* */

Comment (rurban) [reply test/packettest.c:15]: @slontis: Agreed. Will do in some next commit

**Comment by hlandau** [comment]
Agree trivial.

**Comment by t8m** [comment]
Merged to master, 3.1, 3.0 branches. Thank you for your contribution.

**Pull request closed without merging**